tests/pyfunc/test_model_export_with_loader_module_and_data_path.py

```diff
@@ -786,6 +788,106 @@ def test_log_model_persists_requirements_in_mlflow_model_directory(
     assert pyfunc_custom_env_dict["dependencies"][-1]["pip"] == requirements
 
 
+@pytest.mark.large
+def test_check_requirements_and_local_installed_mismatch():
+    import cloudpickle
+    import mlflow.utils.requirements_utils
+
+    with TempDir() as tmp_dir:
+        model_path = tmp_dir.path()
+        req_file_path = tmp_dir.path("requirements.txt")
+        with open(req_file_path, "w") as fp:
+            fp.write(
+                f"cloudpickle=={cloudpickle.__version__}\nscikit-learn=={sklearn.__version__}\n"
+            )
+
+        with mock.patch("mlflow.pyfunc._logger.warning") as mock_warning:
+            # Test case for all packages satisfying requirements.
+            _warn_dependency_requirement_mismatches(model_path)
+            mock_warning.assert_not_called()
+
+            mock_warning.reset_mock()
+
+            original_get_installed_version_fn = (
+                mlflow.utils.requirements_utils._get_installed_version
+            )
+
+            def gen_mock_get_installed_version_fn(mock_versions):
+                def mock_get_installed_version_fn(package, module=None):
+                    if package in mock_versions:
+                        return mock_versions[package]
+                    else:
+                        return original_get_installed_version_fn(package, module)
+
+                return mock_get_installed_version_fn
+
+            # Test multiple mismatched errors printed
+            with mock.patch(
+                "mlflow.utils.requirements_utils._get_installed_version",
+                gen_mock_get_installed_version_fn(
+                    {
+                        "scikit-learn": "999.99.11",
+                        "cloudpickle": "999.99.22",
+                    }
+                ),
+            ):
+                _warn_dependency_requirement_mismatches(model_path)
+                mock_warning.assert_called_once()
+                warning_msg = mock_warning.call_args_list[0][0][0]
+                assert warning_msg.startswith(
+                    "Detected one or more mismatches between the model's dependencies "
+                    "and the current Python environment"
+                )
+                assert (
+                    "scikit-learn (current: 999.99.11, required: "
+                    f"scikit-learn=={sklearn.__version__}" in warning_msg
+                )
+                assert (
+                    " - cloudpickle (current: 999.99.22, required: "
+                    f"cloudpickle=={cloudpickle.__version__})" in warning_msg
+                )
```

Comment:
Can we combine these assertions?

Reply:
then it make a ling too long which is hard to format

Reply:
```suggestion
                assert warning_msg.startswith(
                    """
Detected one or more mismatches between the model's dependencies and the current Python environment:
  - scikit-learn current: 999.99.11, required: scikit-learn=={sklearn_version}"
  - cloudpickle (current: 999.99.22, required: "cloudpickle=={cloudpickle_version})
""".format(sklearn_version=sklearn.__version__, cloudpickle_version=cloudpickle.__version__)
)
```

Can we use `"""`?

Reply:
This might cause a pylint error but we can disable it.

Reply:
I change it to be
```
            mock_warning.assert_called_once_with(
                "Detected one or more mismatches between the model's dependencies and the current "
                "Python environment:\n - cloudpickle (current: 999.99.22, required: "
                "cloudpickle==2.0.0)\n - scikit-learn (current: 999.99.11, required: "
                "scikit-learn==1.0.2)"
            )
```
which is easier for the intent formatting.

Reply:
but it's harder to read.